core/templates/dev/head/components/forms/schema_editors/SchemaBasedListEditorDirective.js

```diff
@@ -135,12 +135,23 @@ oppia.directive('schemaBasedListEditor', [
             }
           };
 
+          var _deleteElementIfEmpty = function() {
+            for (var i = 0; i < $scope.localValue.length - 1; i++) {
+              if ($scope.localValue[i].length === 0) {
+                $scope.deleteElement(i);
+                i--;
+              }
+            }
+          };
+
           $scope.lastElementOnBlur = function() {
             _deleteLastElementIfUndefined();
+            _deleteElementIfEmpty();
             $scope.showAddItemButton();
           };
 
           $scope.showAddItemButton = function() {
+            _deleteElementIfEmpty();
```

Comment:
I suspect I know what's going on here. Currently the directive has on-input-blur="showAddItemButton" (basically) -- and I'm guessing you're changing that to on-input-blur="showAddItemButton($index)". Note that these are different: in the first, you're passing a function reference (that isn't executed yet), and in the second you are actually executing a function and passing the result in. Thus, if you've been doing it like this, it might be the case that you're executing the function more often than you're expecting, which results in the behaviour you're seeing. Is that correct, or did you do something different?

In any case, changing how the function is passed in is tricky and will require more edits to the schema based editor, so based on the technical difficulty there, I think it's fine to keep this function the way you have it. I recommend resolving the UI issues with @jacobdavis11 and getting his agreement, and then we can merge this.